src/caffe/layers/window_data_layer.cpp

```diff
@@ -281,7 +281,7 @@ void WindowDataLayer<Dtype>::InternalThreadEntry() {
       if (this->cache_images_) {
         pair<std::string, Datum> image_cached =
           image_database_cache_[window[WindowDataLayer<Dtype>::IMAGE_INDEX]];
-        cv_img = DecodeDatumToCVMatNative(image_cached.second);
+        cv_img = DecodeDatumToCVMat(image_cached.second, true);
```

Comment:
This doesn't seem right -- `ImageData` has its own color / not color flag that should be respected https://github.com/BVLC/caffe/blob/dev/src/caffe/proto/caffe.proto#L522... or maybe it should do the native decoding unless `is_color` is set to true?